src/eos-metrics-instrumentation.c

```diff
@@ -296,17 +287,14 @@ is_human_session (GVariant *session_parameters)
  * humanity_by_session_id set. Otherwise, return FALSE.
  */
 static gboolean
-add_session_to_set (GVariant *session_parameters)
+add_session_to_set (const char *session_id)
 {
-    if (!is_human_session (session_parameters))
+    if (!is_human_session (session_id))
       return FALSE;
 
-    gchar *session_id;
-    g_variant_get_child (session_parameters, 0, "s", &session_id);
     G_LOCK (humanity_by_session_id);
-    g_datalist_set_data (&humanity_by_session_id, session_id, (gpointer) TRUE);
+    g_datalist_set_data (&humanity_by_session_id, session_id, GUINT_TO_POINTER (TRUE));
```

Comment:
Not sure if anyone already shared these with you:
* https://github.com/endlessm/eos-documentation/wiki/C-Coding-Style
* https://github.com/endlessm/eos-documentation/wiki/JavaScript-Coding-Style